internal/apis/certmanager/types_certificate.go

```diff
@@ -146,14 +146,22 @@ type CertificateSpec struct {
 	// 50 minutes after it was issued (i.e. when there are 10 minutes remaining until
 	// the certificate is no longer valid).
 	//
+	// In addition to an absolute duration, the value can also be expressed as a relative
+	// percentage. For example, if a certificate is valid for 60 minutes, and
+	// `renewBefore=25%`, cert-manager will begin to attempt to renew the certificate
+	// 45 minutes after it was issued (i.e. when there are 15 minutes (25%) remaining
+	// until the certificate is no longer valid).
+	//
 	// NOTE: The actual lifetime of the issued certificate is used to determine the
 	// renewal time. If an issuer returns a certificate with a different lifetime than
 	// the one requested, cert-manager will use the lifetime of the issued certificate.
 	//
 	// If unset, this defaults to 1/3 of the issued certificate's lifetime.
 	// Minimum accepted value is 5 minutes.
-	// Value must be in units accepted by Go time.ParseDuration https://golang.org/pkg/time/#ParseDuration.
-	RenewBefore *metav1.Duration
+	// Duration values must be in units accepted by Go time.ParseDuration
+	// https://golang.org/pkg/time/#ParseDuration. Percent values must be non-negative
+	// integers followed by the percent symbol.
+	RenewBefore *string
```

Comment:
suggestion: We discussed this in the cert-manager open standup on 2024-05-14 and I think the consensus among maintainers was that we're a little worried about the breaking change this represents in the cert-manager Go API, which would break anyone currently setting RenewBefore when creating a Certificate programmatically in Go.

The fix for that break is obviously very simple, but we're generally super cautious about breaking changes and we explicitly support [importing the apis package in Go](https://cert-manager.io/docs/contributing/importing/#when-you-might-import-cert-manager).

The obvious non-breaking alternative is to this is to add a new field, say `RenewBeforePercentage`, which is mutually exclusive with the existing `RenewBefore` field. Do you think that's an approach we could take instead?

Reply:
Sure, I'm happy to remake the change with the new `RenewBeforePercentage` field, that solves the problem equally well for me

Reply:
Ok, pushed a new version introducing a new `RenewBeforePercentage` field